### PR TITLE
feat: database role v1 readiness

### PR DESCRIPTION
**Pull request by sfc-gh-jcieslak**
## Changes
- Checked and updated SDK for database roles (implementation, validation, unit tests, int tests)
- Fixed makefile command name for cleaning models
- Generate asserts and models for database roles and used them in acceptance tests
- Fixed template spacing (it was failing for database role)
- Minor changes in the database role resource + added show_output field
- Moved global variable `id` in acceptance test to local scope (`alerts_acceptance_test.go`)
- Fixed import id and slightly updated example configuration
- Data source implementation and tests adjusted

### CODE REVIEW

File: pkg/resources/database_role.go [hunk 1]
```diff
@@ -63,96 +77,106 @@ func DatabaseRole() *schema.Resource {
...
+				Summary:  "Failed to query database. Marking the resource as removed.",
+				Detail:   fmt.Sprintf("database role name: %s, Err: %s", id.FullyQualifiedName(), err),
+			},
+		}
 	}
 
-	if err := d.Set(FullyQualifiedNameAttributeName, id.FullyQualifiedName()); err != nil {
-		return err
+	databaseRole, err := client.DatabaseRoles.ShowByID(ctx, id)
+	if err != nil && errors.Is(err, sdk.ErrObjectNotFound) {
```
Comment (sfc-gh-jmichalak) [inline pkg/resources/database_role.go:102]: When this error is of different type, we skip this block and continue.

File: pkg/resources/database_role.go [hunk 2]
```diff
@@ -19,38 +21,50 @@ var databaseRoleSchema = map[string]*schema.Schema{
 	"name": {
 		Type:             schema.TypeString,
 		Required:         true,
-		Description:      "Specifies the identifier for the database role.",
 		ForceNew:         true,
```
Comment (sfc-gh-jmichalak) [inline pkg/resources/database_role.go:24]: We can support renames

**Review by sfc-gh-jmichalak — CHANGES_REQUESTED** [review]

File: docs/resources/database_role.md [hunk 3]
```diff
@@ -5,20 +5,25 @@ description: |-
   
 ---
 
+!> **V1 release candidate** This resource was reworked and is a release candidate for the V1. We do not expect significant changes in it before the V1. We will welcome any feedback and adjust the resource if needed. Any errors reported will be resolved with a higher priority. We encourage checking this resource out before the V1 release.
```
Comment (sfc-gh-asawicki) [inline docs/resources/database_role.md:8]: what about such a comment in the deatasource docs?

File: pkg/acceptance/bettertestspoc/README.md [hunk 4]
```diff
@@ -1,4 +1,19 @@
 # Better tests poc
+
+<!-- TOC -->
```
Comment (sfc-gh-asawicki) [inline pkg/acceptance/bettertestspoc/README.md:3]: +1

File: pkg/acceptance/bettertestspoc/config/model/warehouse_model_gen.go [hunk 5]
```diff
@@ -75,6 +76,11 @@ func (w *WarehouseModel) WithEnableQueryAcceleration(enableQueryAcceleration str
 	return w
 }
 
+func (w *WarehouseModel) WithFullyQualifiedName(fullyQualifiedName string) *WarehouseModel {
```
Comment (sfc-gh-asawicki) [inline pkg/acceptance/bettertestspoc/config/model/warehouse_model_gen.go:79]: Just a question: do we have in the generator docs to omit fields that are computed only? (it does not make sense to have fully qualified in the model, so asking to make sure we have it documented to remove it in the future)

File: pkg/internal/genhelpers/generator.go [hunk 6]
```diff
@@ -134,6 +134,7 @@ func generateAndSaveForAllObjects[T ObjectNameProvider, M GenerationModel](objec
 		}
 		filename := filenameProvider(s, model)
 		if err := WriteCodeToFile(&buffer, filename); err != nil {
+			log.Println("Failed to save: ", buffer.String())
```
Comment (sfc-gh-asawicki) [inline pkg/internal/genhelpers/generator.go:137]: you can run the generator with dry-run and verbose options (check here: https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/pkg/internal/genhelpers/README.md#how-it-works) so this log is not necessary

**Review by sfc-gh-asawicki — CHANGES_REQUESTED** [review]

File: pkg/sdk/testint/database_role_integration_test.go [hunk 7]
```diff
@@ -10,6 +10,8 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+// TODO: add tests for new show filtering option and set/unset tags
```
[the remark this answers is not in the document]
Comment (sfc-gh-jcieslak) [reply pkg/sdk/testint/database_role_integration_test.go:13]: 🤦 removed

File: pkg/sdk/database_role_impl.go [hunk 8]
```diff
@@ -84,14 +88,20 @@ func (s *AlterDatabaseRoleRequest) toOpts() *alterDatabaseRoleOptions {
...
 	if s.set != nil {
 		opts.Set = &DatabaseRoleSet{s.set.comment}
 	}
 	if s.unset != nil {
 		opts.Unset = &DatabaseRoleUnset{true}
 	}
+	if s.setTags != nil {
+		opts.SetTags = s.setTags
+	}
+	if s.unsetTags != nil {
```
[the remark this answers is not in the document]
Comment (sfc-gh-jcieslak) [reply pkg/sdk/database_role_impl.go:102]: removed those checks, only `set` and `unset` need them

File: docs/resources/database_role.md (hunk 3 above)
Comment (sfc-gh-jcieslak) [reply docs/resources/database_role.md:8]: Added

File: pkg/internal/genhelpers/generator.go (hunk 6 above)
Comment (sfc-gh-jcieslak) [reply pkg/internal/genhelpers/generator.go:137]: Yeah, my bad forgot to remove this.

File: pkg/resources/database_role.go [hunk 9]
```diff
@@ -21,7 +21,6 @@ var databaseRoleSchema = map[string]*schema.Schema{
 	"name": {
 		Type:             schema.TypeString,
 		Required:         true,
-		ForceNew:         true,
 		Description:      "Specifies the identifier for the database role.",
```
Comment (sfc-gh-jmichalak) [inline pkg/resources/database_role.go:24]: Use `blocklistedCharactersFieldDescription`

Comment (sfc-gh-jcieslak) [reply pkg/resources/database_role.go:24]: Hmm, I thought it was introduced in ur pr because I was looking for it, but could find it. I'll add it.

**Review by sfc-gh-asawicki — APPROVED** [review]

**Review by sfc-gh-jmichalak — APPROVED** [review]

**Review by sfc-gh-asawicki — APPROVED** [review]

**Pull request merged**